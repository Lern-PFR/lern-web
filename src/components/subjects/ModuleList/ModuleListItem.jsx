import PropTypes from 'prop-types';
import styled from 'styled-components';
import { withTranslation } from 'react-i18next';

import routes from 'routes/keys';
import { moduleAccessLinkButton, moduleDescription, moduleListItem, moduleName } from 'theme/pages/subjects/subjectDetailsPage';
import { StyledListItem } from 'components/shared/styledElements';
import { BodyCopy, DoublePica } from 'components/shared/typography';
import { Link } from 'components/shared/navigation';
import { PrimaryLinkButton } from 'components/shared/buttons';

const StyledModuleListItem = styled(StyledListItem)(
	{
		'&:not(:first-child):before': {
			content: '""',
			border: 'solid 1px black',
			position: 'absolute',
			top: '-2.1em',
			height: '2em',
			left: '10em',
		},
	}
);

/**
 * @name ModuleListItem
 * @description An item used to display an overview of a module inside its subject's details page.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {string}	id			The current module's id.
 * @param {string}	name		The current module's name.
 * @param {string}	description	A short description of the current module.
 * @param {bool}	[disabled]	Set to true if the current module is not yet unlocked for the current user.
 * @param {func}	t			The translation method provided by the withTranslation HoC.
 */
const ModuleListItem = ({ id, name, description, disabled, t }) => (
	<StyledModuleListItem {...moduleListItem}>
		<DoublePica {...moduleName}>{name}</DoublePica>
		<BodyCopy {...moduleDescription}>{description}</BodyCopy>
		<PrimaryLinkButton {...moduleAccessLinkButton} disabled={disabled}>
			<Link to={routes.modules.moduleDetails.replace(':id', id)} disabled={disabled}>{t('subjects.links.module')}</Link>
		</PrimaryLinkButton>
	</StyledModuleListItem>
);

ModuleListItem.propTypes = {
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
	disabled: PropTypes.bool,
	t: PropTypes.func.isRequired,
};

ModuleListItem.defaultProps = {
	disabled: false,
};

export default withTranslation()(ModuleListItem);
