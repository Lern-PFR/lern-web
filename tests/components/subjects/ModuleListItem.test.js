import { shallow } from 'enzyme';
import ModuleListItem from 'components/subjects/ModuleList';

describe('ModuleListItem', () => {
	it('should match previous snapshot', () => {
		const sut = (<ModuleListItem id="abcd" name="abcd" description="dummy description" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous disabled-state snapshot', () => {
		const sut = (<ModuleListItem id="abcd" name="abcd" description="dummy description" disabled />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
