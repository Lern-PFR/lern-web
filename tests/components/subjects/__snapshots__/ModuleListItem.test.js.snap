// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleListItem should match previous disabled-state snapshot 1`] = `
<styled.ul
  display="grid"
  gridAutoRows="minmax(250px, 1fr)"
  gridRowGap="2em"
  listStyle="none"
/>
`;

exports[`ModuleListItem should match previous snapshot 1`] = `
<styled.ul
  display="grid"
  gridAutoRows="minmax(250px, 1fr)"
  gridRowGap="2em"
  listStyle="none"
/>
`;
